features/steps/tabs.py

```diff
@@ -0,0 +1,79 @@
+# encoding: utf-8
+
+"""
+Step implementations for paragraph-related features
+"""
+
+from behave import given, then, when
+
+from docx import Document
+from docx.text.tabstop import TabStop
+from docx.shared import Inches
+
+from helpers import test_docx
+
+
+# given ===================================================
+@given('a tab_stops having {count} tab stops')
+def given_a_tab_stops_having_count_tab_stops(context, count):
+    count_int = {
+        '0': 0,
+        '3': 1
+    }[count]
+    document = Document(test_docx('tab-stops'))
+    paragraph_format = document.paragraphs[count_int].paragraph_format
+    context.tab_stops = paragraph_format.tab_stops
+
+
+# when ====================================================
+
+@when('I add a tab_stop')
+def when_i_add_a_tab_stop(context):
+    tab_stops = context.tab_stops
+    tab = tab_stops.add_tab()
+    tab.position = Inches(1.75)
+
+
+@when('I delete a tab_stop')
+def when_i_delete_a_tab_stop(context):
+    tab_stops = context.tab_stops
+    tab = tab_stops[1]
+    tab.delete()
```

Comment:
Hi Steve,

I believe I got the "del tab_stop" mechanism working properly, but I've 
run across an issue while creating unit tests that I think reveals a 
problem that I don't know how to handle.

If I have two tabs, I can delete one or the other and get the 
appropriate expected_xml.  But if I only have ONE tab and I delete it, I 
get 'w:pPr/w:tabs' as the resulting xml rather than 'w:pPr', which I 
believe is what the API specifies.  That is, if there are NO tabs, we 
should not be getting a w:tabs element at all.

Any quick ideas on the proper remedy for this?

David

-- 
David K. Woods, Ph.D.
Researcher, Lead Transana Developer
Wisconsin Center for Education Research
University of Wisconsin, Madison
http://www.transana.org